### PR TITLE
Switch operator to use protobuf client for non-CR resources #250

### PR DESCRIPTION
**Pull request by mfojtik**
This builds on [openshift/library-go#233](https://github.com/openshift/library-go/pull/233) and switches the controller loops that deal with non-CR resources to client-go with protobuf enforced.

/cc @deads2k
/cc @smarterclayton

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -128,22 +128,28 @@ func RunOperator(ctx *controllercmd.ControllerContext) error {
 		ctx.EventRecorder,
 	)
 
-	certRotationController, err := certrotationcontroller.NewCertRotationController(kubeClient, operatorClient, kubeInformersForNamespaces, ctx.EventRecorder)
+	certRotationController, err := certrotationcontroller.NewCertRotationController(
+		kubeClient.CoreV1(),
```
Comment (deads2k) [inline pkg/operator/starter.go]: we want to use a cached one here, right?

Comment (mfojtik) [reply pkg/operator/starter.go]: yes, when library-go portion merge and I get it in bump

**Comment by mfojtik** [comment]
seems to depend on https://github.com/openshift/cluster-kube-controller-manager-operator/pull/154

**Comment by mfojtik** [comment]
/retest

installer flake

**Comment by deads2k** [comment]
surprised this isn't conflicting, but sure

/lgtm

**Pull request merged**